app/controllers/admins_controller.rb

```diff
@@ -43,20 +44,23 @@ def update
 
   # DELETE /admins/1
   def destroy
-    @admin.destroy
+    admin.destroy
     respond_to do |format|
       format.html { redirect_to admins_path, notice: "admin: #{@admin.email} was successfully deleted." }
     end
   end
 
   private
   # Use callbacks to share common setup or constraints between actions.
-  def set_admin
-    @admin = Admin.find(params[:id])
+  def set_admin(admin_entity = nil)
+    @admin = admin_entity || Admin.find(params[:id])
```

Comment:
can we just do seperate action instead of this merged method?
a before action for edit and update
and a before action for new / create